src/pages/recipes/[recipeId].tsx

```diff
@@ -1,14 +1,147 @@
 import { Loader } from "@/shared/components/Loader";
 import { trpc } from "@/utils/trpc";
 import { useRouter } from "next/router";
+import Image from "next/image";
+import { RiShareForwardFill } from "react-icons/ri";
+import { SnackbarProvider } from "@/components/Snackbar";
 
 export default function Recipe() {
   const router = useRouter();
   const { data, isError, isLoading } = trpc.recipes.getRecipe.useQuery({
     recipeId: String(router.query.recipeId),
   });
+
   if (isLoading) {
     return <Loader />;
   }
-  return <div>Unfinished</div>;
+
+  return (
+    <div className="flex flex-col items-center pb-10">
+      <div className="w-2/3 space-y-6">
+        <div className="text-center text-4xl font-bold">{data!.name}</div>
+        <div>
+          {data!.description}
```

Comment:
I think it's generally a good idea to avoid using the ```!``` operator in typescript because it tells the ts to ignore that ```data``` can be null or undefined. In this case, data can be null or undefined if the user changes the recipeId in the url bar to something that doesn't exist and will cause an error.

A possible solution would be to add another if statement to check if data is null after the isLoading if and display a message about invalid recipe id. After this check is added, ts will know that data can't be null or undefined so you can remove the ```!``` operator.

Reply:
Fixed.